x/programs/engine/config.go

```diff
@@ -0,0 +1,316 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package engine
+
+import (
+	"github.com/ava-labs/avalanchego/utils/units"
+
+	"github.com/bytecodealliance/wasmtime-go/v14"
+)
+
+type CompileStrategy uint8
+
+const (
+	// CompileWasm will compile the wasm module before instantiating it.
+	CompileWasm CompileStrategy = iota
+	// PrecompiledWasm accepts a precompiled wasm module serialized by an Engine.
+	PrecompiledWasm
+)
+
+var (
+	DefaultMaxWasmStack         = 256 * units.MiB             // 256 MiB
+	DefaultLimitMaxMemory       = uint32(18 * 64 * units.KiB) // 18 pages
+	DefaultSIMD                 = false
+	DefaultEnableReferenceTypes = false
+	DefaultEnableBulkMemory     = false
+	DefaultProfilingStrategy    = wasmtime.ProfilingStrategyNone
+	DefaultMultiValue           = false
+	DefaultCompileStrategy      = CompileWasm
+
+	defaultWasmThreads                  = false
+	defaultFuelMetering                 = true
+	defaultWasmMultiMemory              = false
+	defaultWasmMemory64                 = false
+	defaultCompilerStrategy             = wasmtime.StrategyCranelift
+	defaultEpochInterruption            = true
+	defaultNaNCanonicalization          = "true"
+	defaultCraneliftOptLevel            = wasmtime.OptLevelSpeed
+	defaultEnableCraneliftDebugVerifier = false
+	defaultEnableDebugInfo              = false
+)
+
+// NewConfig creates a new engine config with default settings
+func NewConfig() *Config {
+	return &Config{
+		inner: DefaultWasmtimeConfig(),
```

Comment:
yeah I guess `inner` is a pattern used in Rust. going to prefix it wasm and instead of Inner() use Get(). so in this case it ill be wasmConfig. wasmEngine etc.